src/fabric/src/fabric2_fdb.erl

```diff
@@ -1328,7 +1339,20 @@ doc_to_fdb(Db, #doc{} = Doc) ->
 
     DiskAtts = lists:map(fun couch_att:to_disk_term/1, Atts),
 
-    Value = term_to_binary({Body, DiskAtts, Deleted}, [{minor_version, 1}]),
+    ValueTerm = case WrappedKEK of
+        false ->
+            {Body, DiskAtts, Deleted};
+        _ ->
+            UpdateCounter = get_stat(Db, <<"update_count">>),
```

Comment:
that we fetch this stat and use it without guaranteeing it is unique is a serious problem. the update happens elsewhere.

Reply:
stats are using atomic operations, if the counter got updated since the start of this transaction it'll get aborted here and retried, so counter guaranteed to be unique in a single transaction.

I have a question though: using read-write on atomic ops kills its performance benefits - can we use uuid instead of update counter, is it cryptographically solid enough? that would allow us to avoid "external" monitonic dependancy all together

Reply:
That would not have a guarantee of uniqueness, which is the only reason we have "update counter" instead of "rev" from the original proposal.